sklearn/cluster/birch.py

```diff
@@ -324,10 +324,23 @@ def radius(self):
 class Birch(BaseEstimator, TransformerMixin, ClusterMixin):
     """Implements the Birch clustering algorithm.
 
-    Every new sample is inserted into the root of the Clustering Feature
-    Tree. It is then clubbed together with the subcluster that has the
-    centroid closest to the new sample. This is done recursively till it
-    ends up at the subcluster of the leaf of the tree has the closest centroid.
+    It is a memory-efficient, online-learning algorithm provided as an alternative
+    to MiniBatchKMeans. It constructs a tree data-structure with the cluster
+    centroids being read off the leaf. These can be either the final cluster
+    centroids or can be provided as input to another clustering algorithm
+    such as `AgglomerativeClustering`.
+
+    The tree data-structure consists of nodes with each node consisting of
+    a number of subclusters. The maximum number of subclusters in a node
+    is determined by the branching factor. Each subcluster maintains a
+    linear sum, squared sum and the number of samples in that subcluster.
+    In addition, each subcluster can also have a node as its child, if the
+    subcluster is not a member of a leaf node.
+
+    For a new point entering the root, it is merged with the subcluster closest
+    to it and the linear sum, squared sum and the number of samples of that
+    subcluster are updated. This is done recursively till the properties of
+    the leaf node are updated.
```

Comment:
The guidelines for docstrings is that the text before the parameters should be short. The reason is that users might not realize that parameters are described after.

I think that the right thing would be to move part of the text that you added in the "notes" section.